docs/src/hardware-schema-typescript-plugin/index.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+var PrebuildPlugin = require("prebuild-webpack-plugin");
+const fs = require("fs");
+const { compileFromFile } = require("json-schema-to-typescript");
+
+function generateHardwareMetadataTypescript() {
+  compileFromFile("../schema/hardware-metadata.schema.json")
+    .then((ts) => fs.writeFileSync("src/hardware-metadata.d.ts", ts))
+    .catch((e) => console.error(e));
+}
```

Comment:
It looks like you can return a promise from the `build` function: https://github.com/hashicorp/prebuild-webpack-plugin/blob/d4dc160fcbe9ef5a66068169331bbf1c0969cc2f/index.js#L84

```suggestion
function generateHardwareMetadataTypescript() {
  return compileFromFile("../schema/hardware-metadata.schema.json")
    .then((ts) => fs.writeFileSync("src/hardware-metadata.d.ts", ts))
    .catch((e) => console.error(e));
}
```
or
```suggestion
async function generateHardwareMetadataTypescript() {
  try {
    const ts = await compileFromFile("../schema/hardware-metadata.schema.json");
    fs.writeFileSync("src/hardware-metadata.d.ts", ts));
  } catch (e) {
    console.error(e);
  }
}
```

Also, is `console.error()` sufficient on failure, or do we actually want to throw an exception though? If throwing an exception is better;

```suggestion
async function generateHardwareMetadataTypescript() {
  const ts = await compileFromFile("../schema/hardware-metadata.schema.json");
  fs.writeFileSync("src/hardware-metadata.d.ts", ts));
}
```

Reply:
Indeed. Much cleaner. Updated. Thanks!